File: common/src/main/java/dev/number6/slack/CallResponse.java
package dev.number6.slack;

public class CallResponse {

    private final boolean isSuccess;
    private final String value;

    public CallResponse(String value) {
        this.isSuccess = true;
        this.value = value;
    }

    public CallResponse(Exception e) {
        this.isSuccess = false;
        this.value = e.getMessage();
    }

    public String body() {
        return value;
    }

    public boolean isSuccess() {
        return isSuccess;
    }
}

File: slack/src/main/kotlin/dev/number6/slack/model/CallResponse.kt
package dev.number6.slack.model

internal class CallResponse {
    val isSuccess: Boolean
    private val value: String

    constructor(value: String) {
        isSuccess = true
        this.value = value
    }

    constructor(e: Exception) {
        isSuccess = false
        value = e.message ?: "No Exception Message."
    }

    fun body(): String {
        return value
    }

}